### ISSUE TITLE
Clean up login and register screens

### ISSUE THREAD

**Issue by m-deacon** [rehive/wallet-react-native#55]
Clean up login and register screens

Suggestion: Let’s reduce the spacing between form title and form input. This will give us more room when signup up.
Suggestion: Thinking “Log in” and “Sign up” should be top right in form sub. Instead of using a button.
Signup button changes to Register on the signup flow.

**Comment by Cupofassam** [comment]
Quick note here: in this issue we added show/hide password to register form:

https://github.com/rehive/wallet-react-native/issues/38

Changes in this issue seem to have overridden those changes on the register page:

https://github.com/rehive/wallet-react-native/issues/18

**Comment by Cupofassam** [comment]
Hey Matt :) Works really well! Particularly like the remembering the last login, super handy.

All mainly seems to work, just a few oddities here and there and a styling tweak.

### Landing page

1. The only thing I would do on this page is add some spacing under the "Version" text - it looks a little cramped underneath. The spacing above and below the  text should be equal.

![image](https://user-images.githubusercontent.com/22153546/39073045-ba5466ba-44ec-11e8-840b-f17fe0854aaf.png)

### Login page

1. Sometimes on the login screen the login button will stick right under a selected field and sometimes not. When it's stuck there it makes it a bit difficult to scroll. Should we maybe make the button stuck to the bottom and not sticky/follow?

Examples:

![whatsapp image 2018-04-20 at 22 28 58](https://user-images.githubusercontent.com/22153546/39073519-5a85879e-44ee-11e8-84f6-eb66877d429a.jpeg)

![whatsapp image 2018-04-20 at 22 28 58 1](https://user-images.githubusercontent.com/22153546/39073535-61ed5552-44ee-11e8-9251-beb36c88f84d.jpeg)

![whatsapp image 2018-04-20 at 22 28 57](https://user-images.githubusercontent.com/22153546/39073547-6af305fc-44ee-11e8-91ac-a7e7e821132e.jpeg)

![whatsapp image 2018-04-20 at 22 28 57 4](https://user-images.githubusercontent.com/22153546/39073558-76487996-44ee-11e8-8565-4da6ba6ee1d3.jpeg)

2. When the company name is invalid, the error it displays is just "Invalid credentials, Unable to log in with provided credentials". It should rather give a direct error on the company field.

![whatsapp image 2018-04-20 at 22 28 57 2](https://user-images.githubusercontent.com/22153546/39073708-facf2d9a-44ee-11e8-8c50-dc8ed190b528.jpeg)

### Register page

1. One of the things we had implemented on the register page recently was centering fields on a user's screen when they were active, which was quite useful to select the next field. Users can obviously use the next button on their keyboards but I think it's good to have multiple ways to easily navigate.

![whatsapp image 2018-04-20 at 22 28 57 3](https://user-images.githubusercontent.com/22153546/39073606-a25758b8-44ee-11e8-9bad-5b45f5d64d18.jpeg)

2. I registered successfully with invalid numbers +27111111111, but with weird behavior. The first time I tap register, it scrolls up to the mobile number field and makes it active, but shows no error. When I tap register again, it registers successfully. However, it did not save the number I entered. When I looked at my settings there was nothing there under mobile numbers. 

3. It gave an error when the number was too short - +27111 - but the error cut off:

![whatsapp image 2018-04-20 at 23 25 45](https://user-images.githubusercontent.com/22153546/39074560-38e9c146-44f2-11e8-9bed-32a29bccc6fb.jpeg)

**Comment by Cupofassam** [comment]
Oh and I'm not sure if it's just my fingers but I feel like I have to tap register twice sometimes. Occasionally it goes to a field. For example, I entered an email that was already registered and tapped register. The form zoomed up to the field but displayed no error. I tapped register again and then the error displayed correctly.

**Comment by m-deacon** [comment]
Cool - going to write this comment as I go and hopefully cover everything:

Landing page:
1. Spacing added

Login page: 
1. I think I fixed this - the buttons weren't placed in the scroll view as they are in register. I didn't get this issue as my phone's screen is very large. Tested on a iPhone 5s simulator and it seems to work now :) I also didn't have automatic scrolling for invalid email etc - added it from register screen.
2. I discussed this with Josh today. The response from the api does not point to which field is incorrect and gives only that error. This is a security thing, but with regards to the company name it may be worth asking for a platform feature to at least tell us that. But that'll need to be discussed.

Register page:
1. Hmm. So I had actually implemented this when clicked "next" you'll see it scrolls. I'm not quite sure how I'll do it now - looked at it and couldn't get it right. Will have another go at it in the morning.
2. As I said, google's library takes anything more than 4 digits in the line number as correct. That's why "+27 11111" or longer will be accepted. I fixed that weird behaviour - taught me something I didn't know about react native that's pretty much core to everything haha. Should be working well now. :)
3. The cut off text is the one issue I still have with this. I'm not sure why, but it's rendering off screen. I'm going to ask Taariq for help on this on Monday.
